### PR TITLE
Support filtering experiments by owner

### LINKED ISSUE #863: Filter experiments by what I own and what I don't own
### What is the expected behavior?

The `GET /experiments` API now supports filtering experiments by `owner=me` and `owner=!me` to only see experiments that I own and don't own, respectively.

If we follow the pattern from PR #858 then I'd suggest adding kwargs like `exclude_mine` and `only_mine` where they can't both be True.

### PR DESCRIPTION
**Pull request by mriedem**
### Summary

This adds the `exclude_mine` and `mine_only` filters
to list experiments that I do or do not own.

### Details and comments

Closes #863

### CODE REVIEW

File: test/ibmq/test_experiment.py [hunk 1]
```diff
@@ -291,6 +291,56 @@ def test_experiments_with_public_filters_error(self):
...
+        # to explicitly create separately-owned epxeriments. We should be able to assume that
+        # there is at least one experiment owned by another user in the integration test
+        # environment though.
+        my_exp = self._create_experiment()
+        not_my_experiments = self.provider.experiment.experiments(exclude_mine=True)
+        # The experiment we just created should not be in the set.
+        not_mine_experiment_uuids = []
+        for experiment in not_my_experiments:
+            self.assertNotEqual(
+                experiment.owner, my_exp.owner,
```
Comment (mriedem) [inline test/ibmq/test_experiment.py:307]: Need to expose `owner` attribute on the Experiment class:

>test/ibmq/test_experiment.py:307:34: E1101: Instance of 'Experiment' has no 'owner' member (no-member)

**Review by mriedem — COMMENTED** [review]
Looks like there is an unrelated mypy failure maybe due to a new version of mypy that was released on Jan 22:

>qiskit/providers/ibmq/experiment/experimentservice.py:347: error: Incompatible types in assignment (expression has type "Tuple[str, int]", variable has type "Union[str, ResultQuality]")  [assignment]

I'll open a separate issue and PR for that.

**Comment by mriedem** [comment]
Currently blocked by #867.

File: test/ibmq/test_experiment.py (hunk 1 above)
Comment (mriedem) [reply test/ibmq/test_experiment.py:307]: Actually I'm not sure why this is complaining:

https://github.com/Qiskit/qiskit-ibmq-provider/blob/84898f924fb053c1d032efe7917eb9a99bda57e8/qiskit/providers/ibmq/experiment/experiment.py#L179

File: test/ibmq/test_experiment.py [hunk 2]
```diff
@@ -291,6 +291,56 @@ def test_experiments_with_public_filters_error(self):
...
+        # there is at least one experiment owned by another user in the integration test
+        # environment though.
+        my_exp = self._create_experiment()
+        not_my_experiments = self.provider.experiment.experiments(exclude_mine=True)
+        # The experiment we just created should not be in the set.
+        not_mine_experiment_uuids = []
+        for experiment in not_my_experiments:
+            self.assertNotEqual(
+                experiment.owner, my_exp.owner,  # pylint: disable=no-member
+                'Experiment should not be returned with exclude_mine filter: %s' %
```
Comment (jyu00) [inline test/ibmq/test_experiment.py:308]:
```suggestion
                'My experiment should not be returned with exclude_mine filter: %s' %
```

File: test/ibmq/test_experiment.py [hunk 3]
```diff
@@ -291,6 +291,56 @@ def test_experiments_with_public_filters_error(self):
...
+                experiment)
+            not_mine_experiment_uuids.append(experiment.uuid)
+        self.assertNotIn(
+            my_exp.uuid, not_mine_experiment_uuids,
+            'Not my experiment returned with exclude_mine filter: %s' %
```
Comment (jyu00) [inline test/ibmq/test_experiment.py:313]:
```suggestion
            'My experiment returned with exclude_mine filter: %s' %
```

**Review by jyu00 — COMMENTED** [review]
LGTM

**Review by jyu00 — APPROVED** [review]

**Pull request merged**